climada/entity/exposures/crop_production.py

```diff
@@ -218,7 +250,20 @@ def set_from_single_run(self, input_dir=INPUT_DIR, filename=None, hist_mean=HIST
 
         # set historic mean, its latitude, and longitude:
         hist_mean_dict = dict()
-        if isdir(hist_mean):
+        # if hist_mean is given as np.ndarray or dict,
+        # code assumes it contains hist_mean as returned by the hazard crop_potential
+        # however structured in dictionary as hist_mean_dict, with same
+        # bbox extensions as the exposure:
+        if isinstance(hist_mean, dict):
+            if not ('firr' in hist_mean.keys() or 'noirr' in hist_mean.keys()):
+                LOGGER.error('Invalid hist_mean provided: {hist_mean}')
```

Comment:
the Logger prefers to be lazy.

```suggestion
                LOGGER.error('Invalid hist_mean provided: %s' %hist_mean)
```